### PR TITLE
Add/Edit /// comments. Fix some code smell.

### PR DESCRIPTION
**Pull request by JonathanFingold**
Can `QnAMakerDialog2` be renamed to something more meaningful, perhaps `AdaptiveQnaMakerDialog`?

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerDialog2.cs [hunk 1]
```diff
@@ -9,125 +9,165 @@
...
         [JsonConstructor]
         public QnAMakerDialog2([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
             : base(sourceFilePath, sourceLineNumber)
         {
         }
 
         /// <summary>
-        /// Gets or sets the KnowledgeBase Id of your QnA Maker KnowledgeBase.
+        /// Gets or sets the path for storing and retrieving the
+        /// QnA Maker knowledge base ID to query.
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerDialog2.cs:42]: No, these descriptions are not right.
We should use: "the KnowledgeBaseId or an expression which evaluates to a KnowledgeBaseId."
It's not the path.

Comment (JonathanFingold) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerDialog2.cs]: I've updated the comments for all of the xExpression type properties.

**Comment by vipeketi** [comment]
remove obsolete metadataBoost from QnA Maker queries #3723

**Comment by JonathanFingold** [comment]
> remove obsolete metadataBoost from QnA Maker queries #3723

@vipeketi , This PR fixes a documentation issue. The work to remove the `MetadataBoost` property is more involved an touches multiple other classes and tests and should be done in a separate PR.

**Pull request merged**